webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -926,38 +924,78 @@ func (v validator) validateUpdatesWhenPoweredOn(ctx *context.WebhookRequestConte
 	if !equality.Semantic.DeepEqual(vm.Spec.Bootstrap, oldVM.Spec.Bootstrap) {
 		allErrs = append(allErrs, field.Forbidden(specPath.Child("bootstrap"), updatesNotAllowedWhenPowerOn))
 	}
-	if !equality.Semantic.DeepEqual(vm.Spec.Network, oldVM.Spec.Network) {
-		allErrs = append(allErrs, field.Forbidden(specPath.Child("network"), updatesNotAllowedWhenPowerOn))
-	}
 
 	// TODO: More checks.
 
 	return allErrs
 }
 
-func (v validator) validateImmutableFields(_ *context.WebhookRequestContext, vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+func (v validator) validateImmutableFields(ctx *context.WebhookRequestContext, vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
 	var allErrs field.ErrorList
 	specPath := field.NewPath("spec")
 
 	allErrs = append(allErrs, validation.ValidateImmutableField(vm.Spec.ImageName, oldVM.Spec.ImageName, specPath.Child("imageName"))...)
 	allErrs = append(allErrs, validation.ValidateImmutableField(vm.Spec.ClassName, oldVM.Spec.ClassName, specPath.Child("className"))...)
 	allErrs = append(allErrs, validation.ValidateImmutableField(vm.Spec.StorageClass, oldVM.Spec.StorageClass, specPath.Child("storageClass"))...)
 	allErrs = append(allErrs, validation.ValidateImmutableField(vm.Spec.MinHardwareVersion, oldVM.Spec.MinHardwareVersion, specPath.Child("minHardwareVersion"))...)
-	// TODO: More checks.
 
-	// TODO: Allow privilege?
+	allErrs = append(allErrs, v.validateImmutableReserved(ctx, vm, oldVM)...)
+	allErrs = append(allErrs, v.validateImmutableNetwork(ctx, vm, oldVM)...)
 
-	var (
-		oldResourcePolicyName string
-		newResourcePolicyName string
-	)
+	return allErrs
+}
+
+func (v validator) validateImmutableReserved(_ *context.WebhookRequestContext, vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+	var allErrs field.ErrorList
+
+	var newResourcePolicyName, oldResourcePolicyName string
+	if reserved := vm.Spec.Reserved; reserved != nil {
+		newResourcePolicyName = reserved.ResourcePolicyName
+	}
 	if reserved := oldVM.Spec.Reserved; reserved != nil {
 		oldResourcePolicyName = reserved.ResourcePolicyName
 	}
-	if reserved := vm.Spec.Reserved; reserved != nil {
-		newResourcePolicyName = reserved.ResourcePolicyName
+
+	p := field.NewPath("spec", "reserved")
+	return append(allErrs, validation.ValidateImmutableField(newResourcePolicyName, oldResourcePolicyName, p.Child("resourcePolicyName"))...)
+}
+
+func (v validator) validateImmutableNetwork(ctx *context.WebhookRequestContext, vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+	var allErrs field.ErrorList
+
+	oldNetwork := oldVM.Spec.Network
+	newNetwork := vm.Spec.Network
+
+	if oldNetwork == nil && newNetwork == nil {
+		return allErrs
+	}
+
+	// The mutation webhook will add interfaces.
+	if ctx.IsPrivilegedAccount {
```

Comment:
The mutation webhook isn't running in the context of the privileged account. It would be the caller's account. It would only be a privileged account if it was the VM Op service account, cluster-admin, or in the priv account list. The mutation webhook for adding a default NIC only runs on Create, not update, so do we even need this check anyway?